File: apps/api/backend/schemas/order.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class OrderCreateRequest(BaseModel):
    cart_id: UUID
    coupon_name: str | None = None


class OrderItemResponse(BaseModel):
    order_item_id: UUID
    product_id: UUID
    product_name: str
    quantity: int
    unit_price: Decimal
    line_total: Decimal
    currency: str


class OrderCreateResponse(BaseModel):
    order_id: UUID
    user_id: UUID
    cart_id: UUID
    order_status: str
    subtotal_amount: Decimal
    discount_amount: Decimal
    total_amount: Decimal
    currency: str | None = None
    coupon_name: str | None = None
    ordered_at: datetime
    items: list[OrderItemResponse]
    message: str

File: apps/web/src/types/order.ts
export type OrderItem = {
  order_item_id: string;
  product_id: string;
  product_name?: string;
  quantity: number;
  unit_price: string | number;
  discount_amount?: string | number;
  final_item_amount?: string | number;
  line_total?: string | number;
  currency: string;
};

export type OrderHistoryItem = {
  order_history_id: string;
  history_event_type: string;
  history_event_at: string;
  order_id: string;
  user_id: string;
  cart_id?: string;
  coupon_name?: string | null;
  order_status: string;
  payment_id?: string | null;
  payment_status?: string | null;
  subtotal_amount?: string | number;
  discount_amount?: string | number;
  total_amount: string | number;
  currency: string;
  ordered_at?: string;
  created_at?: string;
  updated_at?: string;
  items: OrderItem[];
};

export type OrderHistoryResponse = {
  user_id: string;
  total_orders: number;
  orders: OrderHistoryItem[];
};